routes/api/events.js

```diff
@@ -1,9 +1,376 @@
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    // Create unique code for participant
+    const shortId = shortid.generate();
+
+    const newParticipant = {
+      name: req.body.name,
+      phone: req.body.phone,
+      shortId,
+    };
+
+    try {
+      const event = await Event.findById(req.params.id);
+      const participant = await Event.find({
+        _id: req.params.id,
+        'participants.phone': req.body.phone,
+      });
+
+      // Check if participant exists
+      if (participant.length > 0) {
+        return res.status(200).send('You are already registred');
+      }
+
+      // Add participant to collection and save
+      event.participants.push(newParticipant);
+      await event.save();
+
+      res.status(200).send(shortId);
+    } catch (err) {
+      console.error(err.message);
+      res.status(500).send('Server Error');
+    }
+  }
+);
+
+// @route   POST api/events/status
+// @desc    Check registration status
 // @access  Public
-router.get('/', (req, res) => res.send('Events route'));
+router.post(
+  '/status',
+  [check('shortId', 'Registration code is required').notEmpty()],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    try {
+      const event = await Event.find({
+        'participants.shortId': req.body.shortId,
+      });
+
+      // Check if participant exists
+      if (event.length === 0) {
+        return res.status(404).send('Wrong registration code');
+      }
+
+      const getStatus = event[0].participants.filter((p) => {
+        if (p.shortId === req.body.shortId) {
+          return p;
+        }
+      });
+
+      const status = {
+        name: getStatus[0].name,
+        phone: getStatus[0].phone,
+        shortId: getStatus[0].shortId,
+        status: getStatus[0].status,
+      };
+
+      res.status(200).send(status);
+    } catch (err) {
+      console.error(err.message);
+      res.status(500).send('Server Error');
+    }
+  }
+);
+
+// @route   PUT api/events/id/:id/participants
+// @desc    Edit participant's status
+// @access  Private
+router.put(
+  '/id/:id/participants',
+  [
+    auth,
+    check('status', 'Status is required').notEmpty(),
+    check('shortId', 'Participant id is required').notEmpty(),
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({
+        errors: errors.array(),
+      });
+    }
+
+    try {
+      const participant = await Event.updateOne(
+        {
+          user: req.user.id,
+          _id: req.params.id,
+          'participants.shortId': req.body.shortId,
+        },
+        {
+          $set: { 'participants.$.status': req.body.status },
+        }
+      );
+
+      res.status(200).send('Status changed');

```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/spumony/xvent-project-backend/pullRequest?prid=7531905)
```suggestion
      res.status(200).send("Status changed");
```